Neos.Neos/Classes/Fusion/AbstractMenuItemsImplementation.php

```diff
@@ -76,17 +76,17 @@ public function isCalculateItemStatesEnabled(): bool
     }
 
     /**
-     * Should nodes that have "hiddenInIndex" set still be visible in this menu.
+     * Should nodes that have "hiddenInMenu" set still be visible in this menu.
      *
      * @return boolean
      */
-    public function getRenderHiddenInIndex()
+    public function getRenderHiddenInMenu()
     {
-        if ($this->renderHiddenInIndex === null) {
-            $this->renderHiddenInIndex = (bool)$this->fusionValue('renderHiddenInIndex');
+        if ($this->renderHiddenInMenu === null) {
+            $this->renderHiddenInMenu = (bool)($this->fusionValue('renderHiddenInMenu') ?? $this->fusionValue('renderHiddenInIndex'));
```

Comment:
Do we really want to support the old name, too? Would be cleaner to have a code migration for that as otherwise people will still keep on using the old name and wonder, why there is no corresponding yaml definition.

Reply:
Hi ;) yes good catch ... we do have working migrations regarding eel but fusion property names cannot be that easily migrated ... thus a super slim b/c layer like this might not hurt?

But we should adjust the docs to reference this ;)

Reply:
Is there a way to show an error to developers for this? Sth like:
```
if ($this->fusionValue('renderHiddenInIndex') !== null) {
    throw new Exception(...);
}
```

I know, this has small performance drawbacks, but I would prefer a clean migration personally, as this could lead to confusion in the long run. And the warning can be removed in a later version.

Reply:
Okay yes we could go down this road and additionally consider doing a string replace in all fusion files ;)
Thats a good idea i agree. Wdyt @kitsunet?

Reply:
I wouldn't mind that, I think here was some opposition against throwing errors in fusion implementations as usage can be pretty conditional and you might not notice it before bringing in live.

Reply:
Okay i just removed it without exceptions thrown ^^ lets see how we can migrate this 🙈 ?